lib/node_modules/@stdlib/math/base/special/riemann-zeta/test/test.native.js

```diff
@@ -0,0 +1,318 @@
+		tol = 34.0 * EPS * abs( expected[i] );
+		t.ok( delta <= tol, 'within tolerance. s: '+s[i]+'. v: '+v+'. E: '+expected[i]+'. Δ: '+delta+'. tol: '+tol+'.' );
+	}
+	t.end();
+});
+
+tape( 'the function evaluates the Riemann zeta function for very small values', opts, function test( t ) {
+	var v = zeta( 1.0e-10 );
+
+	// Checked against Julia:
+	t.equal( v, -0.5000000000918938 );
+
+	t.end();
+});
+
+tape( 'if evaluated at a pole (`s = 1`), the function returns `NaN`', opts, function test( t ) {
+	var v = zeta( 1.0 );
+	t.equal( isnan( v ), true, 'returns expected value when provided 1' );
+	t.end();
+});
+
+tape( 'the function returns `1` for all input values greater or equal than `56`', opts, function test( t ) {
+	var s;
+	var v;
+	var i;
+
+	s = linspace( 56.0, 100.0, 200 );
+	for ( i = 0; i < s.length; i++ ) {
+		v = zeta( s[ i ] );
+		t.equal( v, 1.0, 'returns 1 when provided '+s[i] );
+	}
+	v = zeta( PINF );
+	t.equal( v, 1.0, 'returns 1 when provided +infinity' );
+
+	t.end();
+});
+
+tape( 'the function returns `0` for all even negative integers', opts, function test( t ) {
+	var s;
+	var v;
+	var i;
+
+	s = linspace( -2.0, -200.0, 100 );
+	for ( i = 0; i < s.length; i++ ) {
+		v = zeta( s[ i ] );
+		t.equal( v, 0.0, 'returns 0 when provided '+s[i] );
+	}
+	s = -pow( 2.0, 32 ); // |s| is greater than MAX_INT32
+	v = zeta( s );
+	t.equal( v, 0.0, 'returns 0 when provided '+s );
+	t.end();
+});
+
+tape( 'the function handles negative values which are larger in magnitude than the maximum factorial', opts, function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var s;
+	var v;
+
+	// Wolfram: zeta( -170.7 )
+	expected = 4.236821692180446371109004075383326908604561232133963e171;
+
+	s = -170.7;
+	v = zeta( s );
+	delta = abs( v - expected );
+	tol = 53.0 * EPS * abs( expected );
+
+	t.ok( delta <= tol, 'within tolerance. s: '+s+'. v: '+v+'. E: '+expected+' Δ: '+delta+'. tol: '+tol );
+
+	// Wolfram: zeta( -171.1 )
+	expected = 1.762429756041972327545919944532107376580768035147432e172;
+
+	s = -171.1;
+	v = zeta( s );
+	delta = abs( v - expected );
+
+	// Note: FF seems to return less precise results (https://travis-ci.org/math-io/riemann-zeta/jobs/115748766). For Node/Chrome, 286.0*eps.
+	tol = 355.0 * EPS * abs( expected );
+
+	t.ok( delta <= tol, 'within tolerance. s: '+s+'. v: '+v+'. E: '+expected+' Δ: '+delta+'. tol: '+tol );
+
+	t.end();
+});
+
+tape( 'the function handles negative integer values which are larger in magnitude than twice the index of the maximum Bernoulli number', opts, function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var s;
+	var v;
+
+	// Only value satisfies this criterion without overflowing: -259. (|-257|+1)/2| = 128, which is the index of the largest Bernoulli number. -261 overflows.
+
+	// Wolfram: zeta( -259 )
+	expected = 8.760156344622921514904073013488223219302793651253880e306;
+
+	s = -259;
+	v = zeta( s );
+	delta = abs( v - expected );
+	tol = 656.0 * EPS * abs( expected );
```

Comment:
Are you able to confirm the source of why the C implementation requires double the tolerance as the JS implementation?

Reply:
This happens when `s = -259`, which is near to the overflowing value. There is a minor difference in the results for this, where the `c` implementation yields `8.760156344624196e+306` while the `js` implementation gives `8.760156344623199e+306`. I've tried logging intermediate values, checking other functions involved, but couldn't find anything. I'll look more into this.

Reply:
@gunjjoshi Is this related to the compiler settings, as we found in other C implementations? May be worth fiddling with compiler flags to confirm.

Reply:
@kgryte Based on our previous discussion at https://github.com/stdlib-js/stdlib/pull/2298#discussion_r1624765205, I tried this with `CFLAGS="-ffp-contract=off"`. Tests pass with the same tolerances which are present in `test.js`. I've added a comment in `test.native.js`.

Reply:
@gunjjoshi Thank you for investigating!